src/anal.c

```diff
@@ -427,11 +429,65 @@ R_API void r_serialize_anal_blocks_load(R_NONNULL Sdb *db, R_NONNULL RAnal *anal
 	key_parser_free (ctx.parser);
 }
 
+R_API void r_serialize_anal_hint_save(R_NONNULL PJ *j, R_NONNULL RAnalHint *hint) {
```

Comment:
The `RAnalHint` struct is actually one layer on top of the underlying records. Its purpose is to accumulate the hints that affect some address to use it in analysis or display it, but not necessarily directly represent the saved hints underneath, so it is not the optimal source for saving.

I would make the signature of this function something like this:
```c
static void anal_hints_store(R_NONNULL Sdb *db, ut64 addr, R_NONNULL HintsAtAddr *hint)
```
where the `HintsAtAddr` is something like
```c
typedef struct {
	const RVector/*<const RAnalAddrHintRecord>*/ *addr_hints;
	bool arch_set; // true iff there is an arch hint starting at this addr
	const char *arch;
	bool bits_set; // true iff there is a bits hint starting at this addr
	int bits;
} HintsAtAddr;
```
I would then do the full saving loop with something like this:
```c
R_API void r_serialize_anal_hints_save(R_NONNULL Sdb *db, R_NONNULL RAnal *anal) {
	HtUP accumulated = ht_up_new(); // Hash table that maps ut64 -> HintsAtAddr *
	// Collect all hints in the tree to sort them
	r_anal_arch_hints_foreach (a, /* function that adds arch hints to the ht */, &accumulated);
	r_anal_bits_hints_foreach (a, /* function that adds bits hints to the ht */, &accumulated);
	r_anal_addr_hints_foreach (a, /* function that adds addr hints to the ht */, &accumulated);
	... // Finally iterate through all records in the ht and call anal_hints_store() on each
}
```